src/appleseed/renderer/modeling/environmentedf/oslenvironmentedf.cpp

```diff
@@ -182,7 +393,109 @@ namespace
             else value.set(0.0f);
         }
 
+        float compute_pdf(
+            const float             u,
+            const float             v,
+            const float             theta) const
+        {
+            assert(u >= 0.0f && u < 1.0f);
+            assert(v >= 0.0f && v < 1.0f);
+            assert(m_importance_sampler);
+
+            // Compute the probability density of this sample in the importance map.
+            const size_t x = truncate<size_t>(m_importance_map_width * u);
+            const size_t y = truncate<size_t>(m_importance_map_height * v);
+            const float prob_xy = m_importance_sampler->get_pdf(x, y);
+            assert(prob_xy >= 0.0f);
+
+            // Compute the probability density of the emission direction.
+            const float pdf = prob_xy > 0.0f ? prob_xy * m_probability_scale / std::sin(theta) : 0.0f;
+            assert(pdf >= 0.0f);
+
+            return pdf;
+        }
+
+        void build_importance_map(const Project& project, IAbortSwitch* abort_switch)
+        {
+            Stopwatch<DefaultWallclockTimer> stopwatch;
+            stopwatch.start();
+
+            // TODO:
+            // Find importance map width and height from m_inputs (see LatLongMapEnvironmentEDF).
+            m_importance_map_width = 1024;
```

Comment:
You should use the minimum of the importance map's width and some user value with a reasonable default (like 1024).

You can retrieve the user value from the entity's parameter (not the inputs that are bound connections such as colors and textures). You can check how this is done in numerous other entities.